File: SurfClient/app/src/main/java/com/example/surfclient/adapter/SpotAdapter.java
package com.example.surfclient.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.surfclient.R;
import com.example.surfclient.model.Spot;

import java.util.List;

public class SpotAdapter extends RecyclerView.Adapter<SpotHolder> {
    private List<Spot> spotList;

    public SpotAdapter(List<Spot> spotList) {
        this.spotList = spotList;
    }

    @NonNull
    @Override
    public SpotHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.list_spot_item, parent, false);
        return new SpotHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull SpotHolder holder, int position) {
        Spot spot = spotList.get(position);
        holder.destination.setText(spot.getDestination());
        holder.address.setText(spot.getAddress());
        holder.difficulty_level.setText(String.valueOf(spot.getDifficulty_level()));
        holder.photo_url.setText(spot.getPhoto_url());
        holder.surfBreak.setText(spot.getSurfBreak().getDisplayName());
    }

    @Override
    public int getItemCount() {
        return spotList.size();
    }
}

File: kotlinSurf - ancienne version/app/src/main/java/com/example/mobileapp/SpotAdapter.kt
package com.example.mobileapp

import Spot
import SpotRecord
import android.os.Bundle
import android.view.LayoutInflater // Pour "gonfler" le layout XML
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import android.widget.Button
import com.bumptech.glide.Glide

// Adaptateur pour gérer la liste des spots dans le RecyclerView
class SpotAdapter(private val spots: SpotRecord, private val onClick: (Spot) -> Unit) :
    RecyclerView.Adapter<SpotAdapter.SpotViewHolder>() {

    // ViewHolder : Gère la vue individuelle d'un élément de la liste
    //class interne qui represente chaque element individuel de la liste.
    //elle stocke les vues de chaque element pour eviter de rechercher ces vues a chaque iteration.
    //ces donnees sont recuperees dans item_spot.xml avec findViewById et stockees dans des variables.
    class SpotViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val imageView: ImageView = view.findViewById(R.id.imageView)
        val nameTextView: TextView = view.findViewById(R.id.nameTextView)
        val locationTextView: TextView = view.findViewById(R.id.locationTextView)
        val surfBreakTextView: TextView = view.findViewById(R.id.surfBreakTextView)
        val difficultyTextView: TextView = view.findViewById(R.id.difficultyTextView)
        val button: Button = view.findViewById(R.id.button)
    }

    // Créer une nouvelle vue (appelée uniquement quand nécessaire)
    //cette ligne charge item_spot.xml(la mise en page) dans une vue retournee avec SpotViewHolder.
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SpotViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_spot, parent, false)
        return SpotViewHolder(view)
    }

    //methode appelee chaque fois qu'un element de la liste doit etre affiche.
    //elle prend SpotViewHolder et l'index [position] de l'element a afficher.
    //on assigne image, name et location a leur vue correspondante
    override fun onBindViewHolder(holder: SpotViewHolder, position: Int) {
        val spot = spots.records[position]
        holder.nameTextView.text = spot.destination
        holder.locationTextView.text = spot.address
        holder.difficultyTextView.text = View.GONE.toString()
        holder.surfBreakTextView.text = View.GONE.toString()


        val firstPhotoUrl = spot.photos
        // Charger l'image avec Glide
        Glide.with(holder.itemView.context)
            .load(firstPhotoUrl)
            .placeholder(R.drawable.city) // Image par défaut
            .into(holder.imageView)

        // Gestion du clic sur le bouton pour afficher les détails
        holder.button.setOnClickListener {
            onClick(spot)
        }
    }

    //retourne le nombre d'elements dans la liste (ici la taille de la liste)
    override fun getItemCount(): Int = spots.records.size
}
